src/EditorFeatures/Test/SolutionCrawler/WorkCoordinatorTests.cs

```diff
@@ -688,6 +688,38 @@ public void SemanticChange_Propagation()
             }
         }
 
+        [Fact]
+        public void ProgressReporterTest()
+        {
+            var solution = GetInitialSolutionInfoWithP2P();
+
+            using (var workspace = new TestWorkspace(TestExportProvider.CreateExportProviderWithCSharpAndVisualBasic(), SolutionCrawler))
+            {
+                WaitWaiter(workspace.ExportProvider);
+
+                var service = workspace.Services.GetService<ISolutionCrawlerService>();
+                var reporter = service.GetProgressReporter(workspace);
+                Assert.False(reporter.InProgress);
+
+                bool started = false;
+                reporter.Started += (o, a) => { started = true; };
+
+                bool stopped = false;
+                reporter.Stopped += (o, a) => { stopped = true; };
+
+                var registrationService = workspace.Services.GetService<ISolutionCrawlerRegistrationService>();
+                registrationService.Register(workspace);
+
+                workspace.OnSolutionAdded(solution);
+
+                Wait((SolutionCrawlerRegistrationService)registrationService, workspace);
```

Comment:
Can you please add the case of adding more work once the queue is empty again so that we can see start stop start stop?